e2e_test/udf/test.py

```diff
@@ -94,6 +94,47 @@ def jsonb_array_struct_identity(v: Tuple[List[Any], int]) -> Tuple[List[Any], in
     return v
 
 
+ALL_TYPES = "BOOLEAN,SMALLINT,INT,BIGINT,FLOAT4,FLOAT8,DECIMAL,DATE,TIME,TIMESTAMP,INTERVAL,VARCHAR,BYTEA,JSONB"
+
+
+@udf(
+    input_types=ALL_TYPES.split(","),
+    result_type=f"struct<{ALL_TYPES}>",
+)
+def return_all(
+    bool,
+    i16,
+    i32,
+    i64,
+    f32,
+    f64,
+    decimal,
+    date,
+    time,
+    timestamp,
+    interval,
+    varchar,
+    bytea,
+    jsonb,
+):
+    return (
```

Comment:
I added a `print` to inspect the values in python. Despite the e2e test showing values are preserved before and after the call, the values in middle looks incorrect for interval:
```
MonthDayNano(months=-1294967296, days=0, nanoseconds=4294967298)
```

We should also check similarly for java.

Reply:
Java side shows the same value. It looks like an encoding issue in arrow-rs??

Reply:
Maybe. Hex of some values:
`00000000b2d05e00` (3000000000 nanoseconds)
`ffffffffb2d05e00` (-1294967296)
`0000000100000002` (4294967298)

Reply:
🤔
```python
>>> import pyarrow as pa
>>> a = pa.array([(1, 2, 3000000000)], type=pa.month_day_nano_interval())
>>> a.buffers()
[None, <pyarrow.Buffer address=0x22c180301c0 size=16 is_cpu=True is_mutable=True>]
>>> a.buffers()[1].hex()
b'0100000002000000005ED0B200000000'
```